runtime-sdk-macros/src/error_derive.rs

```diff
@@ -170,4 +257,56 @@ mod tests {
 
         crate::assert_empty_diff!(actual, expected);
     }
+
+    #[test]
+    fn generate_error_impl_from() {
+        let expected: syn::Stmt = syn::parse_quote!(
+            const _: () = {
+                use oasis_runtime_sdk::{
+                    self as sdk,
+                    core::types::Error as RuntimeError,
+                    error::{private::AsDynError, Error as _},
+                };
+                impl sdk::error::Error for Error {
+                    fn module_name(&self) -> &'static str {
+                        THE_MODULE_NAME
+                    }
+                    fn code(&self) -> u32 {
+                        match self {
+                            Self::Foo { .. } => 1u32,
+                        }
+                    }
+                    fn source(&self) -> Option<&(dyn sdk::error::Error + 'static)> {
+                        match self {
+                            Self::Foo { 0: source, .. } => {
+                                std::option::Option::Some(source.as_dyn_error())
+                            }
+                        }
+                    }
+                }
+                impl From<Error> for RuntimeError {
+                    fn from(err: Error) -> RuntimeError {
+                        RuntimeError::new(err.module_name(), err.code(), &err.to_string())
+                    }
+                }
+            };
+        );
+
+        let input: syn::DeriveInput = syn::parse_quote!(
+            #[derive(Error)]
+            #[sdk_error(module_name = "THE_MODULE_NAME")]
+            pub enum Error {
+                #[sdk_error(code = 1)]
+                Foo(
+                    #[from]
+                    #[sdk_source]
```

Comment:
It might be cool to use the same helper attrs as thiserror, but it'd require a lot of work not to be in an uncanny valley of feature parity.